app/models/division.rb

```diff
@@ -124,9 +125,14 @@ def parent_division_and_name
   end
 
   def generate_short_name
-    return if short_name.present? && Division.pluck(:short_name).exclude?(self.short_name)
+    # no change to short_name that is already saved and therefore uniq
+    return if self.short_name.present? && self.short_name == self.attribute_in_database(:short_name)
+
+    # accept a short_name that is being provided and is uniq
+    return if self.short_name.present? && Division.pluck(:short_name).exclude?(self.short_name)
 
-    self.short_name = name.parameterize
+    # short_name not provided or provided short_name is not uniq
+    self.short_name ||= name.parameterize
```

Comment:
If we are adding a short name field to the form, might it make sense to actually remove all this and just use Rails' uniqueness validation and have the user pick a unique one, rather than generating a shortname with a nasty UUID on the end?